### PR TITLE
While statement code

### PR DESCRIPTION
**Pull request by DuncanMcLean**
Added while statement to Syntax.hs, Parser.hs, and Lexer.hs

### CODE REVIEW

File: src/Adder/Lang/Lexer.hs [hunk 1]
```diff
@@ -37,7 +37,7 @@ adderLexSpec =
       -- TODO Define the reserved names/keywords for the Adder language
       -- See https://docs.python.org/3/reference/lexical_analysis.html#keywords
       Tok.reservedNames =
-        ["from", "pass", "return", "if", "or", "and", "is", "not", "in", "not in", "is not"],
+        ["from", "pass", "return", "if", "or", "and", "is", "not", "in", "not in", "is not", "while"],
```
Comment (Tientuine) [inline src/Adder/Lang/Lexer.hs:40]: Fine.

File: src/Adder/Lang/Parser.hs [hunk 2]
```diff
@@ -68,9 +68,13 @@ compoundStmt :: IParser Statement
 compoundStmt =
   (choice . map try)
     -- if_stmt ::=  "if" assignment_expression ":" suite   - If statement only
+    -- WhileStmt ::= "while" assignmentExpr ":" suite
     [ IfStmt
         <$> (reserved "if" >> assignmentExpr)
         <*> (reservedOp ":" >> suite)
+      WhileStmt
+        <$> (reserved "while" >> assignmentExpr)
+        <*> (reservedOp ":" >> suite)
```
Comment (Tientuine) [inline src/Adder/Lang/Parser.hs:77]: You need to add a comma at the end of the previous line to separate the rule for IfStmt for your new rule for WhileStmt.

File: src/Adder/Lang/Syntax.hs [hunk 3]
```diff
@@ -25,6 +25,7 @@ data Statement
   | AssignmentStmt Identifier Expression
   | AugmentedAssignmentStmt Identifier AugOp Expression -- Bashir's Augmented Assignment constructor
   | BreakStmt
+  | WhileStmt Expression [Statement]
```
Comment (Tientuine) [inline src/Adder/Lang/Syntax.hs:28]: Good.

**Review by Tientuine — CHANGES_REQUESTED** [review]
One teeny change needed before I can approve.

File: src/Adder/Lang/Parser.hs (hunk 2 above)
Comment (DuncanMcLean) [reply src/Adder/Lang/Parser.hs]: I added the comma to separate the the two statements and pushed it to the branch

**Review by Tientuine — APPROVED** [review]
Looks good.

**Pull request merged**